src/components/BuildBaronAndAnnotations/AddIssueModal.tsx

```diff
@@ -0,0 +1,134 @@
+import React, { useState } from "react";
+import { useMutation } from "@apollo/client";
+import styled from "@emotion/styled";
+import { Body } from "@leafygreen-ui/typography";
+import { Input, Tooltip } from "antd";
+import { ConditionalWrapper } from "components/ConditionalWrapper";
+import { Modal } from "components/Modal";
+import { WideButton } from "components/Spawn";
+import { useBannerDispatchContext } from "context/banners";
+import {
+  AddAnnotationIssueMutation,
+  AddAnnotationIssueMutationVariables,
+} from "gql/generated/types";
+import { ADD_ANNOTATION } from "gql/mutations/add-annotation";
+
+const { TextArea } = Input;
+interface Props {
+  visible: boolean;
+  dataCy: string;
+  closeModal: () => void;
+  taskId: string;
+  execution: number;
+  isIssue: boolean;
+}
+
+export const AddIssueModal: React.FC<Props> = ({
+  visible,
+  dataCy,
+  closeModal,
+  taskId,
+  execution,
+  isIssue,
+}) => {
+  const dispatchBanner = useBannerDispatchContext();
+  const title = isIssue ? "Add Issue" : "Add Suspected Issue";
+  const issueString = isIssue ? "issue" : "suspected issue";
+
+  const [url, setUrlValue] = useState<string>("");
```

Comment:
we should use useReducer instead of useState for the form values. we experimented with useReducer and useState for our spawn page forms and found useReducer made our components simpler to read and write

Reply:
Discussed over slack. Will do in a different PR.